src/analysis/cohort.py

```
@@ -0,0 +1,457 @@
+        core_events_only=True,
+        retention_mode="rolling",
+    )
+
+    # ── 3. Milestones ──
+    milestone_df = extract_milestones(cohort_df, RETENTION_MILESTONES)
+
+    # ── 4. Print summary ──
+    print("\n[Cohort] Monthly cohort retention analysis completed")
+    print("[Cohort] Retention rates by month:")
+
+    observed = milestone_df[milestone_df["observed"] == True]
```

Ruff (0.15.9)

[error] Avoid equality comparisons to `True`; use `milestone_df["observed"]:` for truth checks

Replace with `milestone_df["observed"]`

(E712)